testbed/testbed/mock_backend.go

```diff
@@ -149,6 +159,67 @@ func (mb *MockBackend) ConsumeMetric(md pdata.Metrics) {
 	if mb.isRecording {
 		mb.ReceivedMetrics = append(mb.ReceivedMetrics, md)
 	}
+	if mb.isScraping {
+		currentTimestamp := getMetricTimestamp(md)
+		receivedTimestampsCount := len(mb.ReceivedTimestamps)
+		if receivedTimestampsCount == 0 || !mb.ReceivedTimestamps[receivedTimestampsCount-1].Equal(currentTimestamp) {
+			mb.ReceivedTimestamps = append(mb.ReceivedTimestamps, currentTimestamp)
+		}
+	}
```

Comment:
Please add comments to explain why when scraping this logic needs to be different and how it is different.